curiosrc/alertmanager/task_alert.go

```diff
@@ -0,0 +1,395 @@
+			return
+		}
+
+		_, err = toml.Decode(text, cfg)
+		if err != nil {
+			al.alertMap[Name].err = fmt.Errorf("could not read layer, bad toml %s: %w", layer, err)
+			return
+		}
+
+		for i := range cfg.Addresses {
+			prec := cfg.Addresses[i].PreCommitControl
+			com := cfg.Addresses[i].CommitControl
+			term := cfg.Addresses[i].TerminateControl
+			if prec != nil {
+				for j := range prec {
+					if _, ok := addrMap[prec[j]]; !ok && prec[j] != "" {
+						addrMap[prec[j]] = true
+						uniqueAddrs = append(uniqueAddrs, prec[j])
+					}
+				}
+			}
+			if com != nil {
+				for j := range com {
+					if _, ok := addrMap[com[j]]; !ok && com[j] != "" {
+						addrMap[com[j]] = true
+						uniqueAddrs = append(uniqueAddrs, com[j])
+					}
+				}
+			}
+			if term != nil {
+				for j := range term {
+					if _, ok := addrMap[term[j]]; !ok && term[j] != "" {
+						addrMap[term[j]] = true
+						uniqueAddrs = append(uniqueAddrs, term[j])
+					}
+				}
+			}
+		}
+	}
+
+	var ret string
+
+	for _, addrStr := range uniqueAddrs {
+		addr, err := address.NewFromString(addrStr)
+		if err != nil {
+			al.alertMap[Name].err = xerrors.Errorf("failed to parse address: %w", err)
+			return
+		}
+
+		has, err := al.api.WalletHas(al.ctx, addr)
+		if err != nil {
+			al.alertMap[Name].err = err
+			return
+		}
+
+		if !has {
+			ret += fmt.Sprintf("Wallet %s was not found in chain node. ", addrStr)
+		}
+
+		balance, err := al.api.WalletBalance(al.ctx, addr)
+		if err != nil {
+			al.alertMap[Name].err = err
+		}
+
+		if abi.TokenAmount(MinimumBalance).GreaterThanEqual(balance) {
+			ret += fmt.Sprintf("Balance for wallet %s is below 5 Fil. ", addrStr)
+		}
+	}
+	if ret != "" {
+		al.alertMap[Name].alertString = ret
+	}
+	return
+}
+
+// taskFailureCheck retrieves the count of failed tasks for all machines in the task history table.
+// It populates the alert map with the machine name and the number of failures if it exceeds 10.
+func taskFailureCheck(al *alerts) {
+	Name := "TaskFailures"
+	al.alertMap[Name] = &alertOut{}
+
+	type taskFailure struct {
+		Machine  string `db:"completed_by_host_and_port"`
+		Failures int    `db:"failed_tasks_count"`
+	}
+
+	var taskFailures []taskFailure
+
+	// Get all layers in use
+	err := al.db.Select(al.ctx, &taskFailures, `
+								SELECT completed_by_host_and_port, COUNT(*) AS failed_tasks_count
+								FROM harmony_task_history
+								WHERE result = FALSE
+								AND work_end >= NOW() - $1::interval
+								GROUP BY completed_by_host_and_port
+								ORDER BY failed_tasks_count DESC;`, AlertMangerInterval.Minutes())
+	if err != nil {
+		al.alertMap[Name].err = xerrors.Errorf("getting failed task count for all machines: %w", err)
+		return
+	}
+
+	if len(taskFailures) > 0 {
+		for _, tf := range taskFailures {
+			if tf.Failures > 10 {
+				al.alertMap[Name].alertString += fmt.Sprintf("Machine: %s, Failures: %d. ", tf.Machine, tf.Failures)
+			}
+		}
+	}
+	return
+}
+
+func storageCheck(al *alerts) {
```

Comment:
This one is also quite fuzzy. The best number would be "2 days at max fill speed" which we can't really know. But we can start with a few TB and adjust it as we go. 
We also may want to focus on storage that allows sectors. 
The sealing pipeline should probably get another alert related to dead sectors.

Reply:
I haven't though this one thorough yet. But, yes, plan was to start with only permanent storage path and not sealing paths. Sealing path are covered by reservations. Maybe start with 100 GiB free for each storage path to allow.